enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go

```diff
@@ -148,6 +180,21 @@ func (r *ownResolver) DeleteCodeownersFiles(ctx context.Context, args *graphqlba
 	if err := r.db.Codeowners().DeleteCodeownersForRepos(ctx, repoIDs...); err != nil {
 		return nil, errors.Wrapf(err, "could not delete codeowners file for repos")
 	}
+	a := actor.FromContext(ctx)
+	if a.IsAuthenticated() && !a.IsMockUser() {
+		if err := usagestats.LogBackendEvent(
+			r.db,
+			a.UID,
+			deviceid.FromContext(ctx),
+			"own:ingestedCodeownersFile:deleted",
+			nil,
+			nil,
+			featureflag.GetEvaluatedFlagSet(ctx),
+			nil,
+		); err != nil {
+			r.logger.Warn("Could not log own:ingestedCodeownersFile:deleted")
+		}
+	}
```

Comment:
wonder if it'd make sense to pull this out into a function

```suggestion
func logBackendEvent(ctx Context.Context, logger log.Logger, db database.DB, eventName string) {
	a := actor.FromContext(ctx)
	if !a.IsAuthenticated() {
    	return
	}
		if err := usagestats.LogBackendEvent(
			db,
			a.UID,
			deviceid.FromContext(ctx),
            eventName,
			nil,
			nil,
			featureflag.GetEvaluatedFlagSet(ctx),
			nil,
		); err != nil {
			logger.Warn("Could not log own:ingestedCodeownersFile:deleted")
		}
	}
```

Reply:
👍